sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/optimizer/Optimizer.scala

```diff
@@ -384,9 +374,25 @@ object ColumnPruning extends Rule[LogicalPlan] {
     // Eliminate no-op Projects
     case p @ Project(projectList, child) if sameOutput(child.output, p.output) => child
 
+    // Eliminate no-op Window
+    case w: Window if sameOutput(w.child.output, w.output) => w.child
+
+    // Convert Aggregate to Project if no aggregate function exists
+    case a: Aggregate if !containsAggregates(a.expressions) =>
+      Project(a.aggregateExpressions, a.child)
```

Comment:
I mean, `select a from t group by a` is totally different than `select a from t`